### PR TITLE
pythonPackages.markdown2: 2.3.6 -> 2.4.0

### PR DESCRIPTION
**Pull request by duijf**
###### Motivation for this change

`pythonPackages.markdown2` is out of date. [According to PyPI](https://pypi.org/project/markdown2/#history) `2.3.6` is from September 2018. `2.4.0` is from January 2012.

Here is the changelog: https://github.com/trentm/python-markdown2/blob/master/CHANGES.md

~~NB: The new version has dropped support for Python 2. Not sure if that's a problem. Let me know if you'd like to see anything changed.~~ Nevermind this, I mixed up two sets of release notes.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by duijf** [comment]
I don't understand the [evaluation error](https://gist.github.com/GrahamcOfBorg/aa59aafcd06ff86e7e22a2700c3a0694). This file does not seem to use `stdenv.lib`. Would appreciate instructions :)

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111676` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.json-schema-for-humans</li>
</ul>
</details>
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>fava</li>
<li>json-schema-for-humans (python38Packages.json-schema-for-humans)</li>
<li>python37Packages.json-schema-for-humans</li>
<li>python37Packages.markdown2</li>
<li>python37Packages.ramlfications</li>
<li>python38Packages.markdown2</li>
<li>python38Packages.ramlfications</li>
<li>python39Packages.markdown2</li>
<li>python39Packages.ramlfications</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.markdown2:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.markdown2:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.markdown2:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by duijf** [comment]
@SuperSandro2000 Thanks for the review! I addressed the comments and force-pushed a new commit.

File: pkgs/development/python-modules/markdown2/default.nix [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+    owner = "trentm";
+    repo = "python-markdown2";
+    rev = "${version}";
+    sha256 = "sha256:03qmf087phpj0h9hx111k4r5pkm48dhb61mqhp1v75gd09k0z79z";
   };
 
+  checkInputs = [ pygments ];
+
+  checkPhase = ''
+    python ./test/test.py
```
Comment (jonringer) [inline pkgs/development/python-modules/markdown2/default.nix:18]: to support alternative python interpreters such as pypy
```suggestion
    ${python.interpreter} ./test/test.py
```

File: pkgs/development/python-modules/markdown2/default.nix [hunk 2]
```diff
@@ -1,15 +1,23 @@
-{ lib, buildPythonPackage, fetchPypi }:
+{ lib, buildPythonPackage, fetchFromGitHub, pygments }:
```
Comment (jonringer) [inline pkgs/development/python-modules/markdown2/default.nix:1]:
```suggestion
{ lib, buildPythonPackage, fetchFromGitHub, python, pygments }:
```

File: pkgs/development/python-modules/markdown2/default.nix (hunk 1 above)
Comment (duijf) [reply pkgs/development/python-modules/markdown2/default.nix]: Thanks! I've force pushed a new commit to include your suggestions

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111676` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.json-schema-for-humans</li>
</ul>
</details>
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>fava</li>
<li>json-schema-for-humans (python38Packages.json-schema-for-humans)</li>
<li>maestral-gui</li>
<li>python37Packages.json-schema-for-humans</li>
<li>python37Packages.markdown2</li>
<li>python37Packages.ramlfications</li>
<li>python38Packages.markdown2</li>
<li>python38Packages.ramlfications</li>
<li>python39Packages.markdown2</li>
<li>python39Packages.ramlfications</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.markdown2:
<details>

An avoidable string conversion got detected: `    rev = "${version}";`
Please do not convert variables to a string without modifying them but use them directly instead.

</details>
python38Packages.markdown2:
<details>

An avoidable string conversion got detected: `    rev = "${version}";`
Please do not convert variables to a string without modifying them but use them directly instead.

</details>
python39Packages.markdown2:
<details>

An avoidable string conversion got detected: `    rev = "${version}";`
Please do not convert variables to a string without modifying them but use them directly instead.

</details>

File: pkgs/development/python-modules/markdown2/default.nix [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 
-  src = fetchPypi {
-    inherit pname version;
-    extension = "zip";
-    sha256 = "08a124043aa0ad36ba2136239547d5011a2b770278abb11a5609611e0040ea05";
+  # PyPI does not contain tests, so using GitHub instead.
+  src = fetchFromGitHub {
+    owner = "trentm";
+    repo = "python-markdown2";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/markdown2/default.nix:11]:
```suggestion
    rev = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/markdown2/default.nix (hunk 3 above)
Comment (duijf) [reply pkgs/development/python-modules/markdown2/default.nix]: Thanks! Fixed.

(I did check my changes with `nixpkgs-review` myself but this suggestion didn't show up. I guess I was using an older version that didn't have this check yet)

Comment (SuperSandro2000) [reply pkgs/development/python-modules/markdown2/default.nix]: It is something extra I wrote https://github.com/SuperSandro2000/nixpkgs-review-checks

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111676` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.json-schema-for-humans</li>
</ul>
</details>
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>fava</li>
<li>json-schema-for-humans (python38Packages.json-schema-for-humans)</li>
<li>maestral-gui</li>
<li>python37Packages.json-schema-for-humans</li>
<li>python37Packages.markdown2</li>
<li>python37Packages.ramlfications</li>
<li>python38Packages.markdown2</li>
<li>python38Packages.ramlfications</li>
<li>python39Packages.markdown2</li>
<li>python39Packages.ramlfications</li>
</ul>
</details>

**Pull request merged**